chronos/asyncmacro2.nim

```diff
@@ -304,7 +316,8 @@ template await*[T](f: Future[T]): auto =
     if chronosInternalRetFuture.mustCancel:
       raise newCancelledError()
     chronosInternalTmpFuture.internalCheckComplete()
-    cast[type(f)](chronosInternalTmpFuture).internalRead()
+    when T isnot void:
```

Comment:
`Future[void]` still can has an exception inside, and in such case you swallow all the exceptions inside failed `Future[void]`.

Reply:
`internalCheckComplete` checks for the error, `internalRead` just copies the value (it would possibly make sense to make `internalRead` a template such that a copy can be avoided, but `lent` serves a similar purpose in future Nim versions